src/west/commands/project.py

```diff
@@ -19,13 +20,211 @@
+            update_config('manifest', 'path', manifest_path)
+        finally:
+            shutil.rmtree(tempdir, ignore_errors=True)
+
+        self.topdir = topdir
+
+        return projects, manifest_project.abspath
+
+    def projects(self, manifest_file):
+        try:
+            return Manifest.from_file(manifest_file).projects
+        except MalformedManifest as mm:
+            log.die(mm.args[0])
+        except MalformedConfig as mc:
+            log.die(mc.args[0])
+
+    def fixup_zephyr_base(self, projects):
+        for project in projects:
+            if project.path == 'zephyr':
+                update_config('zephyr', 'base', project.path)
+
+    def create(self, directory, exist_ok=True):
+        try:
+            os.makedirs(directory, exist_ok=exist_ok)
+        except PermissionError:
+            log.die('Cannot initialize in {}: permission denied'.
+                    format(directory))
+        except FileExistsError:
+            log.die('Something else created {} concurrently; quitting'.
+                    format(directory))
+        except Exception as e:
+            log.die("Can't create directory {}: {}".format(directory, e.args))
+
+    def clone_manifest(self, url, rev, dest, exist_ok=False):
+        _msg('Cloning manifest repository from {}, rev. {}'.format(url, rev))
+        if not exist_ok and exists(dest):
+            log.die('refusing to clone into existing location ' + dest)
+
+        subprocess.check_call(('git', 'init', dest))
+        subprocess.check_call(('git', 'remote', 'add', 'origin', '--', url),
+                              cwd=dest)
+        is_sha = _is_sha(rev)
+        if is_sha:
+            # Fetch the ref-space and hope the SHA is contained in
+            # that ref-space
+            subprocess.check_call(('git', 'fetch', 'origin', '--tags',
+                                   '--', 'refs/heads/*:refs/remotes/origin/*'),
+                                  cwd=dest)
+        else:
+            # Fetch the ref-space similar to git clone plus the ref
+            # given by user.  Redundancy is ok, for example if the user
+            # specifies 'heads/master'. This allows users to specify:
+            # pull/<no>/head for pull requests
+            subprocess.check_call(('git', 'fetch', 'origin', '--tags', '--',
+                                   rev, 'refs/heads/*:refs/remotes/origin/*'),
+                                  cwd=dest)
+
+        try:
+            # Using show-ref to determine if rev is available in local repo.
+            subprocess.check_call(('git', 'show-ref', '--', rev), cwd=dest)
+            local_rev = True
+        except subprocess.CalledProcessError:
+            local_rev = False
+
+        if local_rev or is_sha:
+            subprocess.check_call(('git', 'checkout', rev), cwd=dest)
+        else:
+            subprocess.check_call(('git', 'checkout', 'FETCH_HEAD'), cwd=dest)
+
 class PostInit(WestCommand):
```

Comment:
Why do we need this at all? Can't this now be moved to the Init() command?

Reply:
Bad rebase. i meant to remove it.

Reply:
Understood